src/osp/Active/SysVehicle.cpp

```diff
@@ -317,40 +313,73 @@ ActiveEnt SysVehicle::part_instantiate(PrototypePart& part,
             shader.m_lightPosition = Vector3{10.0f, 15.0f, 5.0f};
             shader.m_ambientColor = 0x111111_rgbf;
             shader.m_specularColor = 0x330000_rgbf;
-            m_scene.reg_emplace<Phong::ACompPhongInstance>(currentEnt, std::move(shader));
+            rScene.reg_emplace<Phong::ACompPhongInstance>(currentEnt, std::move(shader));
 
-            CompDrawableDebug& bBocks = m_scene.reg_emplace<CompDrawableDebug>(
+            CompDrawableDebug& bBocks = rScene.reg_emplace<CompDrawableDebug>(
                         currentEnt, meshRes, &Phong::draw_entity, 0x0202EE_rgbf);
 
             //new DrawablePhongColored(*obj, *m_shader, *mesh, 0xff0000_rgbf, m_drawables);
         }
         else if (currentPrototype.m_type == ObjectType::COLLIDER)
         {
-            ACompCollisionShape& collision = m_scene.reg_emplace<ACompCollisionShape>(currentEnt);
+            ACompCollisionShape& collision = rScene.reg_emplace<ACompCollisionShape>(currentEnt);
             const ColliderData& cd = std::get<ColliderData>(currentPrototype.m_objectData);
             collision.m_shape = cd.m_type;
 
         }
     }
 
-
-    // first element is 100% going to be the root object
-
-    // Temporary: add a rigid body root
-    //CompNewtonBody& nwtBody
-    //        = m_scene->get_registry().emplace<CompNewtonBody>(newEntities[0]);
-
     // Create mass
-    m_scene.reg_emplace<ACompMass>(newEntities[0], part.get_mass());
+    rScene.reg_emplace<ACompMass>(newEntities[0], part.get_mass());
 
     // return root object
     return newEntities[0];
 }
 
+void SysVehicle::update_activate(ActiveScene &rScene)
+{
+    ACompAreaLink *pArea = SysAreaAssociate::try_get_area_link(rScene);
+
+    if (pArea == nullptr)
+    {
+        return;
+    }
+
+    Universe &rUni = pArea->get_universe();
+    TypeSatIndex type = rUni.sat_type_find_index<universe::SatVehicle>();
+    ActivationTracker& activations = pArea->get_tracker(type);
+
+    // Delete vehicles that have gone too far from the ActiveArea range
+    for (auto const &[sat, ent] : activations.m_leave)
+    {
+        rScene.hier_destroy(ent);
+    }
+
+    // Update transforms of already-activated vehicle satellites
+    auto view = rScene.get_registry().view<ACompVehicle, ACompTransform,
+                                           ACompActivatedSat>();
+    for (ActiveEnt vehicleEnt : view)
+    {
+        auto &vehicleTf = view.get<ACompTransform>(vehicleEnt);
+        auto &vehicleSat = view.get<ACompActivatedSat>(vehicleEnt);
+        SysAreaAssociate::sat_transform_set_relative(
+            rUni, pArea->m_areaSat, vehicleSat.m_sat, vehicleTf.m_transform);
+    }
+
+    // Activate nearby vehicle satellites that have just entered the ActiveArea
+    for (auto &entered : activations.m_enter)
+    {
+        universe::Satellite sat = entered->first;
+        ActiveEnt &rEnt = entered->second;
```

Comment:
i'd say structured bindinds, but it looks like m_enter is a container of pointers based on this code?

Reply:
I suppose you could write a wrapper class that had iterators that automatically dereferenced for you, to get the nice syntax.

though, i'm imagining that you don't want todo that.

Reply:
Could look something like (there are lots of problems with the below example)
```cpp
template<typename RANGE_T>
struct AutoDerefRange
{
    using underlyingIt_t = decltype(m_range.begin());
    struct Iterator
    {
        Iterator(underlyingIt_t value) : m_value(value) {};
        underlyingIt_t m_value;
        decltype(auto) operator*() { return *(m_value.operator*());
        decltype(auto) operator&() { return &(m_value.operator*());
        operator++() { m_value++; }
    }
    AutoDerefRange(RANGE_T & range) : m_range(range) {}
    Iterator begin() { return m_range.begin(); }
    Iterator end() { return m_range.end(); }
}
```

Reply:
`activations.m_enter` is a `std::vector<unordered_map<Satellite, ActiveEnt>::iterator>`s. It tracks new elements added to `activations.m_inside`, allowing the values to be modified (an ActiveEnt) without needing to access the map again. It might be better as a `std::pair<Satellite, std::reference_wrapper<ActiveEnt>>`.

Reply:
My intention was to say that you could get sexier syntax here at the place you're using the std::vector if you wrote a wrapper class to help get sexier syntax.

Reply:
As for references versus iterators, i'd point out that the same conditions that invalidate an iterator also typically invalidate the thing it's pointing to.

So if a reference makes your life easier....